src/download/tests/download-curl-resume.rs

```diff
@@ -62,36 +36,48 @@ fn partially_downloaded_file_gets_resumed_from_byte_offset() {
 #[test]
 fn callback_gets_all_data_as_if_the_download_happened_all_at_once() {
     let tmpdir = tmp_dir();
-
-    let from_path = tmpdir.path().join("download-source");
-    write_file(&from_path, "xxx45");
-
     let target_path = tmpdir.path().join("downloaded");
     write_file(&target_path, "123");
 
-    let from_url = Url::from_file_path(&from_path).unwrap();
+    let (addr, _shutdown) = serve_file(b"xxx45".to_vec());
+
+    let from_url = format!("http://{}", addr).parse().unwrap();
 
-    let received_in_callback = Arc::new(Mutex::new(Vec::new()));
+    let callback_partial = Mutex::new(false);
+    let callback_len = Mutex::new(None);
+    let received_in_callback = Mutex::new(Vec::new());
 
     download_to_path_with_backend(Backend::Curl,
                                   &from_url,
                                   &target_path,
                                   true,
                                   Some(&|msg| {
+        println!("event: {:?}", msg);
```

Comment:
Woops, yep. Used to learn that I needed to add a `Seek` to the reqwest backend file downloader.